play-services-core/src/main/kotlin/org/microg/gms/accountsettings/ui/MainActivity.kt

```diff
@@ -134,7 +135,7 @@ class MainActivity : AppCompatActivity() {
         Log.d(TAG, "Invoked with ${intent.action} and extras $extras")
         super.onCreate(savedInstanceState)
 
-        val screenId = intent?.getIntExtra(EXTRA_SCREEN_ID, -1).takeIf { it != -1 } ?: ACTION_TO_SCREEN_ID[intent.action] ?: 1
+        val screenId = ACTION_TO_SCREEN_ID[intent.action] ?: intent?.getIntExtra(EXTRA_SCREEN_ID, -1).takeIf { it != -1 } ?: 1
```

Comment:
What is the screen id sent by Google Maps? Giving priority to the action-based screen id will mean that we might not show the appropriate screen in some cases, so maybe there's a better way to handle this.

Reply:
screen id is 1

Reply:
I adjusted the code so that only actions that specify a screen id other than 1 have preference over the app provided screen id. As this only applies to 3 actions, we should be reasonable safe now.